### ISSUE TITLE
'mono-2.0-bdwgc.dll' patched for DnSpy - Unity v. 2019.2.4

### ISSUE THREAD

**Issue by carlosnizer** [0xd4d/dnSpy#1392]
Hi there all!
I needed the mono-2.0-bdwgc.dll file patched
for the unity version '2019.2.4', to debug. 
However, there is no matching here in the repository for that version, only until 2019.1.8 version. 
I tried to compile it myself, but I couldn't. I got stuck now ...
Does anyone already have the patched Dnspy mono-2.0-bdwgc.dll for this version?
Thanks!

**Comment by carlosnizer** [comment]
Could someone confirm if I can get the patch build for unity version 2019.2.4?
In the repository there are only patches until 2019.1.8 and on the page to compile the patch, only have hashes until 2019.2.0 .... I am not able to compile ... Iam wasting time?

**Comment by carlosnizer** [comment]
No kind souls to answer?

**Comment by tehflambo** [comment]
I'm also curious about this - in my case I'm looking for unity 2019.2.17.

**Comment by pardeike** [comment]
Follow the instructions of https://github.com/0xd4d/dnSpy-Unity-mono/blob/master/README.md and post Issues there if you can't build it. I think that supporting unity mono should be a community effort.

**Comment by tehflambo** [comment]
Thanks! I'll try this.

**Comment by pardeike** [comment]
I just tried but wasn't smart enough to figure out the last step of choosing the right mono checkout. :-/

**Comment by Kr4ken-9** [comment]
@tehflambo I managed to build mono-2.0-bdwgc.dll for unity-2019.2.17-mbe. However, I'm not sure if I did it correctly. Maybe you can test and see if it works for me.

I am going to try to build 2019.2.4 too.

I uploaded it to Google Drive because I did not have a better place to upload it
[mono-2.0-bdwgc.dll for unity-2019.2.17-mbe](https://drive.google.com/file/d/1HrJvMUvEXVVUOESImljQ5jPKZiCWrhKj/)

You can see my changes [here](https://github.com/0xd4d/dnSpy-Unity-mono/pull/4).... although it looks like I messed up the pull request pretty bad.

**Comment by pardeike** [comment]
I am in need of a 2019.2.17-mbe too, so I will test it when I have time. @Kr4ken-9 - the instructions were not totally clear to me, could you enlighten me/us in what you did so I can replicate it and thus spread the knowledge?

**Comment by 0xd4d** [comment]
I haven't updated the Unity files yet because it's too much work (takes a very long time). Ask questions if you need help.

**Comment by pardeike** [comment]
Finding the mono branch and the right checkout wasn't clear at all. The time token fetched earlier - how do I exactly use it? I tried to find a release/commit with that date but couldn't.

**Comment by Kr4ken-9** [comment]
@pardeike As far as I can tell there is no 2019.2 branch so I used the [2019.2-mbe branch](https://github.com/Unity-Technologies/mono/tree/unity-2019.2-mbe) (because it is the major version of 2019.2.17)

On the pull request I mentioned how I chose the commit hash. @0xd4d says to pick the commit hash for a merge which is closest to the timestamp of mono.dll. You can check the pull request for a little more depth, however I basically chose the commit hash for the merge which happened two days before the mono.dll timestamp.

Also, I have a followup question for @0xd4d. In the readme it says to get the timestamp for the correct version of the mono.dll (I am assuming the correct version out of the 4 dlls listed right above it) How do you know which is the correct version? And do you use the timestamps for mono-2.0-bdwgc.dll? I could not find a merge close to the mono-2.0-bdwgc.dll timestamp, so I used one for mono.dll. Was I correct to do this?

**Comment by pardeike** [comment]
In my case, I went with the debug output of the game (RimWorld) and it uses the .net472 64bit version and lists unity 2019.2.17 - btw, I tested debugging with your file and it works!

**Comment by Kr4ken-9** [comment]
I'm glad it works! I just realized that I only uploaded the 64bit version.... so it's a good thing RimWorld is 64bit. I found another issue with some helpful information about this:

https://github.com/0xd4d/dnSpy/issues/1108

**Comment by 0xd4d** [comment]
> Also, I have a followup question for @0xd4d. In the readme it says to get the timestamp for the correct version of the mono.dll (I am assuming the correct version out of the 4 dlls listed right above it) How do you know which is the correct version? And do you use the timestamps for mono-2.0-bdwgc.dll? I could not find a merge close to the mono-2.0-bdwgc.dll timestamp, so I used one for mono.dll. Was I correct to do this?

Yes use `mono.dll` if it's the non-mbe branch and `mono-2.0-bdwgc.dll` if it's the -mbe branch.

**Comment by GR0B** [comment]
Is there patched version of 2019.2.4 floating around yet? I am not able to build it myself.

**Comment by Albeoris** [comment]
I have the same request for 2019.3.0. :(

It seems that we need a simpler tool that does not require manual manipulation to build under the desired version.

**Comment by Kr4ken-9** [comment]
@carlosnizer @GR0B I have uploaded files for unity 2019.2.4 [here](https://drive.google.com/drive/folders/1kTvzO491-0jhNrfy1OViwqBeKXSpuKeF?usp=sharing)

You can see how I created these (commit hash etc) on this PR: https://github.com/0xd4d/dnSpy-Unity-mono/pull/5

**Comment by GR0B** [comment]
Thanks @Kr4ken-9

**Comment by OgnistyPoland** [comment]
Have someone mono-2.0-bdwgc.dll but 2019.3.7 version?

**Comment by CAA42069** [comment]
![alt text](https://images.pornpics.com/1280/201804/16/4265856/4265856_039_ce70.jpg)
